pinot-core/src/test/java/com/linkedin/pinot/core/segment/index/creator/SegmentPartitionTest.java

```diff
@@ -220,16 +221,67 @@ public void testStringRangeConversions() {
     }
   }
 
+  /**
+   * Unit test for {@link SegmentPartitionConfig} that tests the following:
+   * <ul>
+   *   <li> Conversion from/to JSON string. </li>
+   *   <li> Function names, values and ranges are as expected. </li>
+   * </ul>
+   * @throws IOException
+   */
+  @Test
+  public void testSegmentPartitionConfig()
+      throws IOException {
+    Map<String, ColumnPartitionConfig> expectedMap = new HashMap<>();
 
+    for (int i = 0; i < 10; i++) {
+      String partitionColumn = "column_" + i;
+      String partitionFunction = "function_" + i;
+
+      int numValues = 5;
+      StringBuilder builder = new StringBuilder();
+      for (int j = 0; j < numValues; j++) {
+        builder.append("[").append(j).append(" ").append(j + 1).append("]");
+        if (j < numValues - 1) {
+          builder.append(ColumnPartitionConfig.PARTITION_VALUE_DELIMITER);
+        }
+      }
+      expectedMap.put(partitionColumn, new ColumnPartitionConfig(partitionFunction, builder.toString()));
+    }
+
+    SegmentPartitionConfig expectedConfig = new SegmentPartitionConfig(expectedMap);
+    SegmentPartitionConfig actualConfig = SegmentPartitionConfig.fromJsonString(expectedConfig.toJsonString());
```

Comment:
will be useful to add a compatibility test as well. Have a json that has some new fields in it, and make sure it is deserialzed correctly.